static/app/utils/discover/discoverQuery.tsx

```diff
@@ -21,8 +22,18 @@ export type TableData = {
   meta?: MetaType;
 };
 
+function shouldRefetchData(prevProps: DiscoverQueryProps, nextProps: DiscoverQueryProps) {
+  return defined(nextProps.miseryKey) && prevProps.miseryKey !== nextProps.miseryKey;
```

Comment:
I'm doing this to trigger the Discover Query API call once the thresholds are updated. Is there a better way to do this?